src/lib/Syntax.hs

```diff
@@ -1098,6 +1097,7 @@ instance Subst Atom where
       where Abs args' () = subst env $ Abs args ()
     BoxedRef b ptr size body -> BoxedRef b' (subst env ptr) (subst env size) body'
         where Abs b' body' = subst env $ Abs b body
+    ProjectElt idxs v -> getProjection (toList idxs) $ substVar env v
```

Comment:
Since `getProjection` is used in multiple places I'm tempted to leave it as-is. Do you feel strongly about this? I could also try to do a fold at every place where `getProjection` is used, I suppose.